src/types.ts

```diff
@@ -0,0 +1,54 @@
+
+export interface UploaderArgs {
+  build?: string                      // Specify the build number manually
+  branch?: string                     // Specify the branch manually
+  dir?: string                        // Directory to search for coverage reports.
+  env?: string                        // Specify environment variables to be included with this build
+  sha?: string                        // Specify the commit SHA mannually
+  file?: string | string[]            // Target file(s) to upload
+  flags: string | string[]                      // Flag the upload to group coverage metrics
+  name?: string                       // Custom defined name of the upload. Visible in Codecov UI
+  parent?: string                     // The commit SHA of the parent for which you are uploading coverage.
+  pr?: string                         // Specify the pull request number mannually
+  token?: string                      // Codecov upload token
+  tag?: string                        // Specify the git tag
+  verbose?: boolean                   // Run with verbose logging
+  rootDir?: string                    // Specify the project root directory when not in a git repo
+  nonZero?: boolean                   // Should errors exit with a non-zero (default: false)
+  dryRun?: boolean // Don't upload files to Codecov
+  slug?: string // Specify the slug manually (Enterprise use)
+  url?: string // Change the upload host (Enterprise use)
+  clean?: boolean // Move discovered coverage reports to the trash
+  feature?: string // Toggle features
+  source?: string // Track wrappers of the uploader
+}
+
+export type UploaderEnvs = NodeJS.Dict<boolean | string | number | undefined>
```

Comment:
```suggestion
export type UploaderEnvs = NodeJS.Dict<string>
```

```suggestion
export type UploaderEnvs = Record<string, string | undefined>
```

Reply:
```suggestion
export type UploaderEnvs = Partial</* … */>
```

Once `--strictOptionalProperties` is available in TypeScript 4.4, this will be the most accurate way to model the shape of environment variables.

Reply:
Just to verify , these changes will work? Or you are showing me the future ?

Reply:
The top two will. They are variations of each other. `NodeJS.ProcessEnv`  would be a subtype of `UploaderEnvs` if it were written as one of the two. Minor adjustments may have to be made in places expecting `UploaderEnvs ` to potentially be `boolean` or `number`.

The bottom one is for the future. The specific inclusion of `undefined` in the type union would invalidate the future `--strictOptionalProperties` flag.